File: 0231-power-of-two/0231-power-of-two.java
class Solution {
    public boolean isPowerOfTwo(int n) {
        if(n==1) return true;
        if(n<=0) return false;

       double x = Math.log10(n)/Math.log10(2);
       int a = (int)Math.floor(x);            
       int b = (int)Math.ceil(x);            

        return a==b;
    }
}

File: 0231-power-of-two/0231-power-of-two.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isPowerOfTwo = function(n) {
    if(n==0) return false;
    let x = (Math.log(n)/Math.log(2)).toFixed(9)
    // console.log(x)
    return x == Math.trunc(x);
};